presto-main/src/test/java/com/facebook/presto/remotetask/TestReactorNettyHttpClient.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.remotetask;
+
+import com.facebook.airlift.http.client.HttpClient.HttpResponseFuture;
+import com.facebook.airlift.http.client.Request;
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.airlift.json.JsonCodecFactory;
+import com.facebook.airlift.json.JsonObjectMapperProvider;
+import com.facebook.presto.execution.TaskStatus;
+import com.facebook.presto.server.InternalCommunicationConfig;
+import com.facebook.presto.server.remotetask.ReactorNettyHttpClient;
+import com.facebook.presto.server.remotetask.ReactorNettyHttpClientConfig;
+import com.facebook.presto.server.smile.AdaptingJsonResponseHandler;
+import com.facebook.presto.server.smile.JsonResponseWrapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.units.Duration;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Mono;
+import reactor.netty.DisposableServer;
+import reactor.netty.http.HttpProtocol;
+import reactor.netty.http.server.HttpServer;
+
+import java.net.URI;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static com.facebook.airlift.http.client.Request.Builder.prepareGet;
+import static com.facebook.presto.execution.TaskState.PLANNED;
+import static com.facebook.presto.server.RequestHelpers.getJsonTransportBuilder;
+import static com.facebook.presto.server.smile.AdaptingJsonResponseHandler.createAdaptingJsonResponseHandler;
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+
+public class TestReactorNettyHttpClient
```

Comment:
This is a great small smoke test to verify `ReactorNettyHttpClient` can send HTTP requests. It would be great to also enable `ReactorNettyHttpClient` for one of the integration tests (for both, Java and Native execution) to make sure `ReactorNettyHttpClient` is properly integrated and when enabled queries run succesfully